core/tests/protractor_utils/forms.js

```diff
@@ -293,34 +300,41 @@ var SetOfHtmlStringEditor = function(elem) {
 var UnicodeEditor = function(elem) {
   return {
     setValue: async function(text) {
-      await elem.element(by.tagName('input')).clear();
-      await elem.element(by.tagName('input')).sendKeys(text);
+      var tagNameInput = elem.element(by.tagName('input'));
+      await action.clear('tag Name Input', tagNameInput);
+      await action.sendKeys('Tag Name Input', tagNameInput, text);
     }
   };
 };
 
 var AutocompleteDropdownEditor = function(elem) {
   return {
     setValue: async function(text) {
-      await elem.element(by.css('.select2-container')).click();
+      var select2ContainerButton = elem.element(by.css('.select2-container'));
+      await action.click('Test Container Button', select2ContainerButton);
       // NOTE: the input field is top-level in the DOM, and is outside the
       // context of 'elem'. The 'select2-dropdown' id is assigned to the input
       // field when it is 'activated', i.e. when the dropdown is clicked.
-      await element(by.css('.select2-dropdown')).element(
-        by.css('.select2-search input')).sendKeys(text + '\n');
+      var select2DropDown = element(by.css('.select2-dropdown')).element(
+        by.css('.select2-search input'));
+      await action.sendKeys('Select2 Drop Down', select2DropDown, text + '\n');
     },
     expectOptionsToBe: async function(expectedOptions) {
-      await elem.element(by.css('.select2-container')).click();
+      var select2ContainerButton = elem.element(by.css('.select2-container'));
+      await action.click('Test Container Button', select2ContainerButton);
       var actualOptions = await element(by.css('.select2-dropdown'))
         .all(by.tagName('li')).map(
           async function(optionElem) {
             return await optionElem.getText();
           }
         );
+      await waitFor.visibilityOf(
+        actualOptions, 'Actual options taking too long to appear');
```

Comment:
`actualOptions` isn't an element. It's a list of strings. Above, the `.map(...)` call iterates over the list of elements, gets each element's text, and returns a list of those texts

Reply:
updated!